aarch64/src/interrupts.rs

```diff
@@ -0,0 +1,90 @@
+use crate::registers::EsrEl1;
+use core::fmt;
+use port::println;
+
+#[cfg(not(test))]
+core::arch::global_asm!(include_str!("interrupts.S"));
+
+pub fn init() {
+    unsafe { init_interrupts() };
+}
+
+extern "C" {
+    fn init_interrupts();
+}
+
+/// Register frame at time interrupt was taken
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct InterruptFrame {
+    x0: u64,
+    x1: u64,
+    x2: u64,
+    x3: u64,
+    x4: u64,
+    x5: u64,
+    x6: u64,
+    x7: u64,
+    x8: u64,
+    x9: u64,
+    x10: u64,
+    x11: u64,
+    x12: u64,
+    x13: u64,
+    x14: u64,
+    x15: u64,
+    x16: u64,
+    x17: u64,
+    x18: u64,
+    frame_pointer: u64, // x29
+    link_register: u64, // x30
+    esr_el1: EsrEl1,
+    elr_el1: u64,
+    far_el1: u64,
+    interrupt_type: u64,
+    xzr: u64, // zero padding
+}
+
+impl fmt::Debug for InterruptFrame {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
Is this substantially different than what you'd get out of `#[derive(Debug)]`?

Reply:
The derived version would write all the numbers in decimal rather than hex.

Reply:
Oh! At the print site you'd just give, `{:x?}` to get hex.

Reply:
I did not know that.

Reply:
Changed